WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -54,28 +54,19 @@ public async Task BuildPrivacySuggestionsAsync(Wallet wallet, TransactionInfo in
 		var loadingRing = new LoadingSuggestionViewModel();
 		Suggestions.Add(loadingRing);
 
-		try
+		var hasChange = transaction.InnerWalletOutputs.Any(x => x.ScriptPubKey != destination.ScriptPubKey);
+
+		if (hasChange && !isFixedAmount && !info.IsPayJoin)
 		{
-			var hasChange = transaction.InnerWalletOutputs.Any(x => x.ScriptPubKey != destination.ScriptPubKey);
+			var suggestions =
+				ChangeAvoidanceSuggestionViewModel.GenerateSuggestionsAsync(info, destination, wallet, linkedCts.Token);
 
-			if (hasChange && !isFixedAmount && !info.IsPayJoin)
+			await foreach (var suggestion in suggestions)
 			{
-				var suggestions =
-					ChangeAvoidanceSuggestionViewModel.GenerateSuggestionsAsync(info, destination, wallet, linkedCts.Token);
-
-				await foreach (var suggestion in suggestions)
-				{
-					Suggestions.Insert(Suggestions.Count - 1, suggestion);
-				}
+				Suggestions.Insert(Suggestions.Count - 1, suggestion);
 			}
 		}
-		catch (OperationCanceledException)
```

Comment:
Is this ever hit on current master? My guess is that it is not. Do I get it right?

Reply:
> Do I get it right?

Yes, we are returning `false` on [cancellation](https://github.com/zkSNACKs/WalletWasabi/blob/640b1aaa180657eb490d8badab2ce3720b0e5972/WalletWasabi/Blockchain/TransactionBuilding/BranchAndBound.cs#L114), then grab the `BestSelectionFound` [here](https://github.com/zkSNACKs/WalletWasabi/blob/640b1aaa180657eb490d8badab2ce3720b0e5972/WalletWasabi/Blockchain/TransactionBuilding/ChangelessTransactionCoinSelector.cs#L81). By throwing an exception we wouldn't reach that line.